### ISSUE TITLE
Consistently use app_name in strings

### ISSUE THREAD

**Issue by TheLastProject** [CatimaLoyalty/Android#2228]
Several places in strings.xml use the direct text "Catima", it should refer to app_name everywhere so that in languages where the name Catima is transliterated it consistently uses the same everywhere.

**Comment by TheLastProject** [comment]
> I suggest changing the value of the key "app_name" in the file located at "Android/app/src/main/res/values-ko/strings.xml" from "Catima" to "카티마." Since I don't know to get editing permissions, I'm leaving this as a comment.

Anyone with a Weblate account can edit translations. If you translate the app name, make sure to translate it everywhere, also in Fastlane, for consistency.

**Comment by shawshank725** [comment]
Android studio doesn't allow concatenating strings in xml file. It allows [referencing single string](https://stackoverflow.com/questions/4746058/reference-one-string-from-another-string-in-strings-xml). Using {{}} or strings/ doesnt work.

This [link ](https://stackoverflow.com/questions/5764548/how-to-concatenate-multiple-strings-in-android-xml)mentions how it can be done. It uses java code to insert the strings at runtime. 
```
<string name="welcome_messages">Hello, %1$s! You have %2$d new messages.</string>
String text = getString(R.string.welcome_messages, username, mailCount);
```
An example from android studio docs.

**Comment by TheLastProject** [comment]
I think I'd much prefer the %1$s (or simply %s) way as this is already what is used in Catima. It just requires changing many many strings and making a few small code changes, which is error prone and takes a lot of time so I haven't bothered with it yet (a regex might help).

**Comment by shawshank725** [comment]
> I think I'd much prefer the %1$s (or simply %s) way as this is already what is used in Catima. It just requires changing many many strings and making a few small code changes, which is error prone and takes a lot of time so I haven't bothered with it yet (a regex might help).

I can help with that. However I would want to change string.xml files in a batch, not all at once, as you already mentioned that this is error prone.

**Comment by TheLastProject** [comment]
Yeah doing it in batches seems better, also reduces the chance of conflicts with translation updates :)

**Comment by shawshank725** [comment]
I noticed that in strings.xml file, some IDs are snake cased and some camel cased. 
I'm trying to refactor the code and it is correct in xml and kt file. But it seems to be giving some error I don't know why.

I have added a new variable called 'CATIMA' that is a string. Used in the code below.

<img width="757" height="215" alt="Image" src="https://github.com/user-attachments/assets/d9a4466c-b55e-41e1-b0e3-11493cbe08c3" />

The first line: `getString(R.string.permission_read_cards_label, CATIMA)` contains this label: `permission_read_cards_label`. Initially it was in camel case: `permissionReadCardsLabel `and giving me error. But then I changed it to `permission_read_cards_label `and the error was gone. Any explanation behind this? 

The changes are made according to this documentation given on [Android Studio docs](https://developer.android.com/guide/topics/resources/string-resource.html#kotlin).

The issue that I'm now encountering is that the app doesn't build because the ID permissionReadCardsLabel is not present.

**Comment by TheLastProject** [comment]
> I noticed that in strings.xml file, some IDs are snake cased and some camel cased.

Yeah it's a bit inconsistent but I recommend just not changing it because it makes the scope of this change unnecessarily big. The different casing in the strings.xml files are not something that causes issues.

> I have added a new variable called 'CATIMA' that is a string. Used in the code below.

We already have the app_name string though, just use that one everywhere.

**Comment by shawshank725** [comment]
So I found out the following: 

1. Android studio has this linting feature to check string IDs. It is OK with camel case string ID but doesn't understand snake cased.
2. When you write: getString(R.string.permissionReadCardsLabel), if the ID is in camel case the linting feature gives an error: `Format string '`permissionReadCardsLabel`' is not a valid format string so it should not be passed to `String.format``
3. This error can be suppressed though.

Still working on it.

**Comment by TheLastProject** [comment]
> Android studio has this linting feature to check string IDs. It is OK with camel case string ID but doesn't understand snake cased.

That doesn't sound correct, both cases have been in use for literal years with no issues.
